### PR TITLE
upgrade Llama-Index integration to 0.10

### PR DESCRIPTION
**Pull request by joshreini1**
TODO:

- [X] Update `quickstart`
- [x] Update `tru_llama` unit tests
- [X] Change imports in `TruLlama` to support 0.10
- [x] Add bridge for llama-index versions < 0.10

### CODE REVIEW

File: trulens_eval/trulens_eval/tru_llama.py [hunk 1]
```diff
@@ -26,42 +26,42 @@
 
 with OptionalImports(messages=REQUIREMENT_LLAMA):
     import llama_index
-    from llama_index.chat_engine.types import AgentChatResponse
-    from llama_index.chat_engine.types import BaseChatEngine
-    from llama_index.chat_engine.types import StreamingAgentChatResponse
-    from llama_index.embeddings.base import BaseEmbedding
-    from llama_index.indices.base import BaseIndex
+from llama_index.core.chat_engine.types import AgentChatResponse
```
Comment (piotrm0) [inline trulens_eval/trulens_eval/tru_llama.py:29]: missing indentation here and in the others

Comment (piotrm0) [reply trulens_eval/trulens_eval/tru_llama.py:29]: (should be under optional imports)

**Comment by piotrm0** [comment]
Do we need to add a package to optional imports requirements?

**Comment by joshreini1** [comment]
> Do we need to add a package to optional imports requirements?

We're good on this front, I've changed the data loading method in quickstart to the simple directory reader.

**Comment by piotrm0** [comment]
There are some failures at "from llama_index import VectorStoreIndex" in various notebooks.

**Review by piotrm0 — APPROVED** [review]

**Comment by jeya-kumar-jk** [comment]
Issue with pip installation, pip installation mentioned as 0.9.18, but it was upgraded 0.10, index spelling is misspelled , Please execute it colab once, it is not working  in colab

**Pull request merged**